### ISSUE TITLE
Add `ostree show` option to list metadata keys

### ISSUE THREAD

**Issue by dbnicholson** [ostreedev/ostree#2487]
We use a couple custom metadata keys at Endless and I often can't remember their names. I currently run `ostree show --raw` and then pick out the keys from the `g_variant_print` output. That's not that fun, but it's doubly not fun on our OS commits where we include the `ostree.sizes` metadata, which is massive.

Anyways, I think it would be nice if `ostree show` had an option to just print the keys in the commit metadata and detached metadata.

**Comment by jlebon** [comment]
Hmm, or maybe we add `-o json`? (This came up recently in https://github.com/ostreedev/ostree/issues/2469 for `ostree admin status` too.)

**Comment by dbnicholson** [comment]
Yeah, that would be nice, too. Although, you then have to handle conversion from GVariant to JSON, which... maybe? In the `ostree admin status` it's more straightforward textual output that you're just going format in a different way. Ensuring that you can serialize commit metadata to JSON looks harder to guarantee. If you look at the [g_variant_print code](https://gitlab.gnome.org/GNOME/glib/-/blob/main/glib/gvariant.c#L2189), it's not exactly trivial.

**Comment by jlebon** [comment]
We do this in rpm-ostree using json-glib's [`json_gvariant_serialize`](https://github.com/coreos/rpm-ostree/blob/5d93072f91e4b336baba1c703f8c47afc64010c2/src/app/rpmostree-builtin-status.cxx#L1198). But yeah, it'd be a new dep for ostree.
